frontend/components/text-classifier/labeling-rules/RuleLabelsDefinition.vue

```diff
@@ -130,23 +131,27 @@ export default {
       return this.dataset.getCurrentLabelingRuleMetrics() || {};
     },
 
-    selectedLabel() {
-      if (this.selectedLabelsVModel !== undefined) {
-        return this.selectedLabelsVModel[0];
+    selectedLabels() {
+      if (this.selectedLabelsVModel.length) {
+        return this.selectedLabelsVModel;
       }
     },
 
     ruleInfo() {
       // TODO: We can improve this
+      const storedRule =
+        this.currentRule &&
+        this.dataset.findRuleByQuery(this.currentRule.query);
+      const storedRuleLabels = storedRule && storedRule.labels;
+      const queryWithLabelsIsStored = _.isEqual(
+        _.sortBy(storedRuleLabels),
+        _.sortBy(this.selectedLabels)
+      );
       if (this.isSaved) {
         return "The rule was saved";
       }
-      if (
-        this.currentRule &&
-        this.selectedLabelsVModel.length &&
-        this.dataset.findRuleByQuery(this.currentRule.query, this.selectedLabel)
-      ) {
-        return "This query with this label are already saved as rule";
+      if (queryWithLabelsIsStored) {
+        return "This query with this labels are already saved as rule";
```

Comment:
The message should be built depending on the number of labels the rule have defined. Should keep the singular form if the number of labels is equal to one (even for multi-label)